bin/dry-run.rb

```diff
@@ -476,8 +475,20 @@ def handle_dependabot_error(error:, dependency:)
 )
 
 fetcher = Dependabot::FileFetchers.for_package_manager($package_manager).new(**fetcher_args)
-$files = if $options[:clone] || always_clone
+$files = if $repo_contents_path
+           if $options[:cache_steps].include?("files") && Dir.exist?($repo_contents_path)
+            puts "=> reading cloned repo from #{$repo_contents_path}"
+           else
+            puts "=> cloning into #{$repo_contents_path}"
+            FileUtils.rm_rf($repo_contents_path)
+           end
            fetcher.clone_repo_contents
+           if $options[:commit]
```

Comment:
Note: the branch is handled in `clone_repo_contents` but `commit` is a dry-run only option so handled here